src/renderer/components/+workloads-pods/pod-details-secrets.tsx

```diff
@@ -5,70 +5,49 @@
 
 import "./pod-details-secrets.scss";
 
-import React, { Component } from "react";
+import React, { useEffect, useState } from "react";
 import { Link } from "react-router-dom";
-import { autorun, observable, makeObservable } from "mobx";
-import { disposeOnUnmount, observer } from "mobx-react";
+import { autorun, observable } from "mobx";
+import { observer } from "mobx-react";
 import { Pod, Secret, secretsApi } from "../../../common/k8s-api/endpoints";
 import { getDetailsUrl } from "../kube-detail-params";
 
 export interface PodDetailsSecretsProps {
   pod: Pod;
 }
 
-@observer
-export class PodDetailsSecrets extends Component<PodDetailsSecretsProps> {
-  @observable secrets: Map<string, Secret> = observable.map<string, Secret>();
+export const PodDetailsSecrets = observer(({ pod }: PodDetailsSecretsProps) => {
+  const [secrets] = useState(observable.map<string, Secret>());
```

Comment:
Because the state tracking is done via mobx. I don't see how else we would use `observable` here. We need some way to persist the instance between rerenders which is why I am using `useState`

Reply:
I guess we don't even need to use observables here. So instead of `secrets.replace` https://github.com/lensapp/lens/pull/4131/files/6672987cd9a248bfab270604b7973ec1044cb745#diff-b4696a544159db998d3f1a489b66fb4e912e719cc420990dadd0bdfd4086b1bbR30

We can do something like this:

```
const [secrets, setSecrets] = useState(new Map());
const secrets = podSecrets.filter(Boolean).map(secret => [secret.getName(), secret])
setSecrets(new Map(secrets);
```

Reply:
Okay I can change that